tests/conftest.py

```diff
@@ -13,12 +15,51 @@
 from httpx import AsyncioBackend
 
 
-@pytest.fixture(params=[pytest.param(AsyncioBackend, marks=pytest.mark.asyncio)])
+try:
+    from httpx.concurrency.trio import TrioBackend
+except ImportError:  # pragma: no cover
+    TrioBackend = None  # type: ignore
+
+
+@pytest.fixture(
+    params=[
+        pytest.param(AsyncioBackend, marks=pytest.mark.asyncio),
+        pytest.param(TrioBackend, marks=pytest.mark.asyncio),
```

Comment:
Shouldn't this be `pytest.mark.trio`? And instead of making `TrioBackend` equal to `None` above can we make the parameters determined within the global scope inside that `try-except` import block?

Reply:
> Shouldn't this be pytest.mark.trio?

Ah, I think this is something that needs a comment as well. The asyncio mark ensures that there will be an event loop setup for tests that use the `backend` fixture. Even tests that run on trio need the event loop because they're actually run in the asyncio threadpool. Again, this is all mostly because we *need* asyncio to be the top-level loop, mostly for uvicorn compatibility reasons.

> And instead of making TrioBackend equal to None above can we make the parameters determined within the global scope inside that try-except import block?

As in, building a `backend_params` list and appending `TrioBackend` to it in the import block? That might end up cleaner, I'll try it out. :-)